control-plane-operator/controllers/hostedcontrolplane/hostedcontrolplane_controller.go

```diff
@@ -248,6 +249,7 @@ func (r *HostedControlPlaneReconciler) registerComponents() {
 		kubevirtcsiv2.NewComponent(),
 		cnov2.NewComponent(),
 		ntov2.NewComponent(),
+		dnsoperatorv2.NewComponent(),
```

Comment:
we should also turn off v1 reconciliation, you'd need to wrap this:
https://github.com/openshift/hypershift/blob/2753dcc8cfb7591a61d64d26766526f844cebffa/control-plane-operator/controllers/hostedcontrolplane/hostedcontrolplane_controller.go#L1283-L1286

with 
```go
if !r.IsCPOV2 {
}
```